### ISSUE TITLE
Getting Error on Expand Function .

### ISSUE THREAD

**Issue by suraj-mane-zellis** [SAP/olingo-jpa-processor-v4#238]
I am using 2.0.0 release of odata jpa with spring boot 3.1.3 version . I have added OneToOne relationship into Location entity .

When I am use Expand , I am getting error 

URL : http://host url/odata/Locations?$expand=Address

Response : {
    "error": {
        "code": null,
        "message": "Could not determine recommended JdbcType for `com.odata.entity.Address`"
    }
}

Sample Code : 

      @Entity
      @Cacheable(false)
      @Table(name = "Location")
     public class Location {

    
    @Id
    @Column(name = "LocationId", nullable = false)
    private Short locationId;
   
    @Column(name = "AddressId")
    private Integer addressId;
    
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "AddressId",referencedColumnName = "AddressId"
            insertable = false, updatable = false)
        private Address address;
//getters and setters

**Comment by wog48** [comment]
I was able to reproduce the error.

Before I explain what happens, I need to give a short introduction. OData allows to specify a type cast e.g. in navigations [to address a derived type ](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_AddressingDerivedTypes) instead the base type given in the definition of the navigation property. As an example in the following test [BusinessPartnerRoles(BusinessPartnerID='98',RoleCategory='X')/BusinessPartner/com.sap.olingo.jpa.Person](https://github.com/SAP/olingo-jpa-processor-v4/blob/8ab6f7031e2b52c7fb5ec71ebb6d20f85dc87318/jpa/odata-jpa-processor/src/test/java/com/sap/olingo/jpa/processor/core/query/TestJPAQueryNavigation.java#L57) can be taken. To support this, a type cast of a JPA From is done in [JPAAbstractJoinQuery#createFromClauseNavigationJoins](https://github.com/SAP/olingo-jpa-processor-v4/blob/8ab6f7031e2b52c7fb5ec71ebb6d20f85dc87318/jpa/odata-jpa-processor/src/main/java/com/sap/olingo/jpa/processor/core/query/JPAAbstractJoinQuery.java#L747):

``` Java    
        final JPAEntityType cast = this.navigationInfo.get(i + 1).getEntityType();
        target = (From<?, ?>) target.as(cast.getTypeClass());
```
In fact this is not only done if a type cast is requested, but always, which should not be a problem. It runs well with various EclipseLink implementation (last tested with 4.0.2)  as well as the own Criteria Builder implementation [odata-jpa-processor-cb](https://github.com/SAP/olingo-jpa-processor-v4/tree/8ab6f7031e2b52c7fb5ec71ebb6d20f85dc87318/jpa/odata-jpa-processor-cb). Looking at OpenJPA implementation I am optimistic that it would work as well. Only Hibernate gave a problem. Here I used version 6.2.7, which came with Spring Boot 3.1.3.

The cast shall be done in SqmCriteriaNodeBuilder#cast. To get the type, the method calls 
`final BasicDomainType<X> type = getTypeConfiguration().standardBasicTypeForJavaType( castTargetJavaType );`

TypeConfiguration#standardBasicTypeForJavaType look as follows:
```Java
		if ( javaType == null ) {
			return null;
		}

		return standardBasicTypeForJavaType(
				javaType,
				javaTypeDescriptor -> new BasicTypeImpl<>(
						javaTypeDescriptor,
						javaTypeDescriptor.getRecommendedJdbcType( getCurrentBaseSqlTypeIndicators() )
				)
		);
```
As _Address_ is an entity the `javaTypeDescriptor` is an instance of EntityJavaType. The implementation of method `getRecommendedJdbcType` always throws an error. 

This looks to me like an error in Hibernate and not in the JPA Processor. Therefor I close the issue.
